### ISSUE TITLE
Specific Drops and Zone Drops behaving like World Drops

### ISSUE THREAD

**Issue by Stifler82** [azerothcore/azerothcore-wotlk#4987]
Originally reported https://github.com/chromiecraft/chromiecraft/issues/286

##### FACTION

- [x] Alliance
- [x] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39

##### CURRENT BEHAVIOUR:

Items that should be very specific Zone Drops and Drops from specific NPCS are in World Drop Loot Tables, also know as: "reference_loot_template.sql" in the database.

Some examples that I've gathered are 
[[Humbert's Helm]](https://wowgaming.altervista.org/aowow/?item=4724) (Dropped off Undead mobs in Duskwood // Should be dropping from Dun'Garok Riflemen in Dun Garok)
[[Blackrock Champion's Axe]](https://wowgaming.altervista.org/aowow/?item=1455) (Dropped off undead mobs in Duskwood // Should be dropping from Blackrock Champion in Redridge)

Other Items that I haven't personally encountered myself but are behaving like World Drops right now and should not:
[[Robes of the Shadowcaster]](https://wowgaming.altervista.org/aowow/?item=1297)
[[Nightwalker Armor]](https://wowgaming.altervista.org/aowow/?item=2234)
[[Cloak of Rot]](https://wowgaming.altervista.org/aowow/?item=4462)

And so many many more... This have to be looked into in-depth.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Zone Drops and Specific Drops from a Specific mob should be dropping from said Zones/Drops respectively.

##### STEPS TO REPRODUCE THE PROBLEM:

1. play the game
2. get drops that don't belong to the area/mob

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/8bacb39d53349f63066e84bcddb5a7f9300acb0d

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by T1ti** [comment]
Kinda related to https://github.com/azerothcore/azerothcore-wotlk/issues/4699

**Comment by Stifler82** [comment]
@Zoidwaffle Can you do this Fix pls?
I trust of your judgment about drop rate.
(And to avoid misunderstanding, and everything you do of course).😉👍

**Comment by FrancescoBorzi** [comment]
@Zoidwaffle are you still planning to work on this?

**Comment by T1ti** [comment]
how is that fixed ? was there any commit that fixed it or is did you give up ?

**Comment by Stifler82** [comment]
![image](https://user-images.githubusercontent.com/1884642/119716057-9b7f1e00-be5c-11eb-9f0e-db3cedb9dfc8.png)

**Comment by T1ti** [comment]
yeah but that's only 1 case .. ? you posted a few as exampels adn there are hundreds of area/zone/map specific drops

**Comment by Stifler82** [comment]
ah ok, my bad, i didn't notice..

**Comment by Stifler82** [comment]
btw, I haven't "Posted" this report, i just only made COPY and PASTE form CC

**Comment by sudlud** [comment]
Also how does this relate to the recent world drop / loot PR

**Comment by Gultask** [comment]
There are some fixes but I can't say it's perfect. I went around fixing some, though.

I don't *think* we have any missing items, but it could be worth checking.
I was using the addon ALLTHETHINGS to find these unique items and checking the DB.
From what I recall the unique drops were correctly assigned to its mob, but then was also located randomly in a reference or two.

**Comment by joshjoshajosh** [comment]
I'm really confused about zone drops, stuff like Unsharpened Ice Razor (50319) I can't find in any table:
`select * from creature_loot_template where item = '50319';
Empty set

select * from reference_loot_template where item = '50319';
Entry | Item  | Comment               |
35072 | 50319 | Unsharpened Ice Razor |

select * from creature_loot_template where reference = '35072';
Empty set`

**Comment by Tereneckla** [comment]
reference_loot_template 35072 is referenced in reference_loot_template 35073 which then is in creature_loot_templates

**Comment by joshjoshajosh** [comment]
So you get different results when you put that in SQL? Because I can't see anything in creature_loot_template with a reference value of '35072', as shows above I get "Empty set"

**Comment by Tereneckla** [comment]
there is 35073 in creature_loot_templates

35072 is in 35073

**Comment by sudlud** [comment]
```sql
select * from creature_loot_template where Reference = '35073';
```

**Comment by joshjoshajosh** [comment]
Ohhhh I understand now thank you, it's like a looped/nested reference the same table.

**Comment by joshjoshajosh** [comment]
Ok just 1 more question, marrowgar's loot template shows:

```
+-------+-------+-----------+--------+---------------+----------+---------+----------+----------+-----------------------------------+
| Entry | Item  | Reference | Chance | QuestRequired | LootMode | GroupId | MinCount | MaxCount | Comment                           |
+-------+-------+-----------+--------+---------------+----------+---------+----------+----------+-----------------------------------+
| 36612 |     1 |     34230 |    100 |             0 |        1 |       0 |        1 |        2 | Lord Marrowgar - (ReferenceTable) |
```

Minimum 1, maximum 2, but it always drops 2 items. Can someone help me understand this behaviour?
